index.html

```diff
@@ -118,17 +238,75 @@
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+
+    if (options.state !== 'all') {
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
-      });
+        return d.state === options.state;
+      }); 
     }
 
     // SCATTERPLOT
 
+
+    var points = svg.selectAll('.point')
+                    .data(txData);
+
+                    points.enter().append('circle')
+                    .attr('class', 'point')
+                    .attr('r', 5);
+
+                    points
+                    .attr('cx', function(d) { return x(d.democratContribution); })
+                    .attr('cy', function(d) { return y(d.republicanContribution)});
+
+                    points.exit().remove();
+
     // HISTOGRAMS
 
+    var barWidth = ( x(x.ticks()[1]) - x(x.ticks()[0]) )/5;
+
+    // x axis histogram
+    var demContr = txData.map(function (i) {return i.democratContribution});
+
+    var xHistogram = d3.layout.histogram()
+                             .bins(x.ticks())
+                             (demContr); 
+
+    histScale = d3.scale.linear()
+                    .domain([0, d3.max(xHistogram, function (d) { return d.y; })])
+                    .range([0, height/5])
+                    .nice();
+
+    var xBars = svg.selectAll(".xbar")
+                  .data(xHistogram)
+                  .enter().append("g")
```

Comment:
Note that because we immediately call `.enter()` after joining the data, the `bars` variable contains just the enter selection. This means that once we have added all of the bars the first time `update()` is called, we'll never change them again — there's always the same number of bars, so nothing to enter, thus all of this code is called on an empty selection the second and subsequent times it runs. We would instead call `bars.enter()...` on a new line so that we can continue to access the full `bars` selection.

Compare and contrast to how `points` is done above.

A group-specific note, although we could just get rid of the groups and create rects directly: we would need to append the rects in the enter selection, when we first create the groups. Otherwise, we'll just keep adding new rects every time through. Then in order to modify the rect inside each of the bar groups, we would use something like: `bars.selectAll('rect').attr(...)`